### PR TITLE
perf: need not narrow the gradients and zero gradients for dnn backend

### PR DESCRIPTION
**Pull request by i8run**
## What changes were proposed in this pull request?

In `DistriOptimizer`, we need not narrow the gradients and zero gradients for dnn backend.

## How was this patch tested?
Tested on VGG-16 training.

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala [hunk 1]
```diff
@@ -315,7 +317,9 @@ object DistriOptimizer {
             (0 until _subModelNumber).map { i =>
               () => {
                 cached.localModels(i).training()
-                cached.localModels(i).zeroGradParameters()
+                if (Engine.getEngineType() != MklDnn) {
+                  cached.localModels(i).zeroGradParameters()
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala]: can we override dnn layer zeroGradParameter with an empty method?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala [hunk 2]
```diff
@@ -94,6 +94,8 @@ class SpatialBatchNormalization(
 
     runningMean.zero()
     runningVariance.zero()
+
+    gradWeightAndBias.zero()
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala]: do not check affine?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialConvolution.scala [hunk 3]
```diff
@@ -368,8 +369,6 @@ class SpatialConvolution(
   }
 
   override def zeroGradParameters(): Unit = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialConvolution.scala]: maybe we can just remove the method. I remember the default impl is do nothing, right?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Linear.scala [hunk 4]
```diff
@@ -279,8 +280,6 @@ class Linear(
   }
 
   override def zeroGradParameters(): Unit = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Linear.scala]: maybe we can just remove the method. I remember the default impl is do nothing, right?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala [hunk 5]
```diff
@@ -303,9 +305,6 @@ class SpatialBatchNormalization(
   }
 
   override def zeroGradParameters(): Unit = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala]: maybe we can just remove the method. I remember the default impl is do nothing, right?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialConvolution.scala (hunk 3 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialConvolution.scala]: No. The default implementation will get parameters and zero gradients.

```scala
  def zeroGradParameters(): Unit = {
    if (parameters() != null) {
      parameters()._1.zip(parameters()._2)foreach{ case (weight, grad) =>
        grad.resizeAs(weight).zero()
      }
    }
  }
```

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala (hunk 2 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/SpatialBatchNormalization.scala]: This BatchNorm only support the true of affine.
